### PR TITLE
Add routing settings

### LINKED ISSUE #344: Add routing settings to the CLI
- [ ] Force HTTPS
- [ ] Sticky session
- [ ] Canonical

### PR DESCRIPTION
**Pull request by EtienneM**
Fix #344

```text
$ scalingo -a biniou force-https [--enable]
-----> Force HTTPS has been enabled on biniou
```

```text
$ scalingo -a biniou force-https --disable
-----> Force HTTPS has been disabled on biniou
```

```text
$ scalingo -a biniou sticky-session [--enable]
-----> Sticky session has been enabled on biniou
```

```text
$ scalingo -a biniou sticky-session --disable
-----> Sticky session has been disabled on biniou
```

```text
$ scalingo -a biniou domains
+-----------------+-----+
|     DOMAIN      | SSL |
+-----------------+-----+
| biniou.fr       |   - |
| example.com (*) |   - |
+-----------------+-----+
  (*) canonical domain
```

```text
$ scalingo -a biniou set-canonical-domain biniou.fr
-----> Canonical domain set to biniou.fr
```

```text
$ scalingo -a biniou unset-canonical-domain
-----> Canonical domain disabled
```

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -3,107 +3,140 @@
...
 [[projects]]
   branch = "master"
+  digest = "1:e25669f49084df10ec9ff820eb625cfd90a136331bfb79112f88307f7829ad9f"
   name = "github.com/Scalingo/go-gitremote"
   packages = ["."]
+  pruneopts = "NUT"
   revision = "97b505dac0c3696d7395d6627494ec105a69280f"
 
 [[projects]]
+  branch = "fix/cli/344/router_settings"
```
Comment (johnsudaar) [inline Gopkg.lock]: User the master branch

File: Gopkg.toml [hunk 2]
```diff
@@ -25,8 +25,9 @@
   name = "github.com/Scalingo/envconfig"
 
 [[constraint]]
-  version = "~1.4"
   name = "github.com/Scalingo/go-scalingo"
+  branch = "fix/cli/344/router_settings"
```
Comment (johnsudaar) [inline Gopkg.toml]: Same here

File: Gopkg.lock (hunk 1 above)
Comment (EtienneM) [reply Gopkg.lock]: Then merge https://github.com/Scalingo/go-scalingo/pull/71 :p

File: cmd/domains.go [hunk 3]
```diff
@@ -145,4 +145,60 @@ var (
...
+	}
+
+	unsetCanonicalDomainCommand = cli.Command{
+		Name:     "unset-canonical-domain",
+		Category: "App Management",
+		Flags:    []cli.Flag{appFlag},
+		Usage:    "Unset a canonical domain.",
+		Description: `Unset the canonical domain of this app:
+
+    $ scalingo -a myapp unset-canonical-domain example.com
```
Comment (johnsudaar) [inline cmd/domains.go]: Why not just `unset-canonical-domain` ? There is only one canonical domain per app.

Comment (EtienneM) [reply cmd/domains.go]: :heavy_check_mark:

**Comment by Soulou** [comment]
@johnsudaar is it good for you? (review says no)

**Comment by johnsudaar** [comment]
Why? In this case shouldn't every actions prefixed by app:
`app:restart` `app:env-set` `app:scale`

Why this one and not the others ?

**Comment by Soulou** [comment]
Hmm, maybe not app, but like `router:force-https` or `routing:force-https`, because stick-session lost in the numerous commands we have, it's not super clear, as we keep adding commands.

**Comment by Soulou** [comment]
It's in the scope of this PR as we're adding some new commands.

**Comment by EtienneM** [comment]
@Soulou Then do you agree if I rename all the argument so that we have:
`env-set` -> `env:set` etc for all commands?

**Comment by Soulou** [comment]
There should be retrocompatibility for EVERYTHING. You don't break habits of people like this.

**Comment by Soulou** [comment]
I completely agree that it's for another PR.

The question is what commands for this PR right now

* `scalingo settings-force-https --enable`
* `scalingo settings-canonical-domain --enable`

Just to have them gathered like in the dashboard.

**Comment by johnsudaar** [comment]
Mmh we could move them in a new category called router, but i don't see the point in prefixing them. But i like the current implementation. I think that it suit the current CLI design.

**Comment by Soulou** [comment]
Ok let's stop this horror story and let's merge this then.

**Pull request merged**